app/controllers/admin/source_files_controller.rb

```diff
@@ -32,12 +32,14 @@ def query_template
       super
         .split(" OR ")
         .push("LOWER(active_storage_blobs.filename) LIKE ?")
+        .push("status = ?")
         .join(" OR ")
     end
 
     def query_values
       array = super
-      array + [array.first]
+      term = array.first
+      array + [term, SourceFile.statuses[term.parameterize(separator: "_")]]
```

Comment:
Do you know if this part of the app is often read/changed? 

I ask because I think this line is a bit complex, and in case we do read it often, I think it would be valuable to maybe extract to more simpler/atomic lines to make debugging/following along the code execution easier. If not, then it's fine!

Reply:
It is calling `super` on [this Administrate method](https://github.com/thoughtbot/administrate/blob/44ad6ce1072fbd897f1faaf7231b3b5a0b92f535/lib/administrate/search.rb#L103), so `array` just looks like: `array = ["%searchterm%", "%searchterm", ..., "%searchterm"]`. Do you think something like:

```
array + [term, db_value_for_status]
```

would be better, where I add a method called `db_value_for_status`? And maybe use "values" instead of "array"?

```
def query_values
  values = super
  term = values.first
  values + [term, db_value_for_status(term)]
end

private

def db_value_for_status(term)
  SourceFile.statuses[term.parameterize(separator: "_")
end
```

Or let me know if you had something else in mind.

Reply:
Yes, I like that! Thanks 😊